src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -26,6 +26,7 @@ public unsafe static uint[] Square(uint[] value)
 
         // Mutable for unit testing...
         private static int SquareThreshold = 32;
+        private static int AllocationThreshold = 4096;
```

Comment:
Do you have a sense for the typical size of numbers used?  Obviously there will always be outliers, but if there's a sweet spot where we could get 80% while not having the stack allocation be too large, that'd be great.  4K (or 8K because it's used in a pair) on the stack is a fairly large chunk.  @KrzysztofCwalina, thoughts?